### PR TITLE
Minor Changes to Git Tutorial for Clarification

### PR DESCRIPTION
**Pull request by TheOmid**
Changes made:

1. Clarify for students that their public key is in the file ending in .pub
2. Add warning to never share your private code
3. Clarify that SSH is passwordless so you won't see a prompt
4. Clarify that MacOS pitfall for key caching is only for the token

### CODE REVIEW

File: docs/setup_git.md [hunk 1]
```diff
@@ -94,11 +94,14 @@ $ ssh-keygen -t ed25519 -C "your_email@example.com"
 Your public key has been saved in /Users/awdeorio/.ssh/id_ed25519.pub
 ```
 
-Copy your SSH public key.  Select and copy the output.  **Your output will be different.**
+Copy your SSH public key (the file ending with .pub).  Select and copy the output.  **Your output will be different.**
```
Comment (awdeorio) [inline docs/setup_git.md:97]:
```suggestion
Copy your SSH public key (the file ending with `.pub`).  Select and copy the output.  **Your output will be different.**
```

File: docs/setup_git.md [hunk 2]
```diff
@@ -94,11 +94,14 @@ $ ssh-keygen -t ed25519 -C "your_email@example.com"
...
 ```console
 $ cat ~/.ssh/id_ed25519.pub
 ssh-ed25519 KLBJDjlkaksfadhinoueliwekljhfasdlkjhfdss/asdnfkjlnaksjdfdfnkljdafslF awdeorio@umich.edu.com
 ```
+<div class="primer-spec-callout danger" markdown="1">
+**WARNING:** Do not share your private key with anyone!
```
Comment (awdeorio) [inline docs/setup_git.md:103]:
```suggestion
**WARNING:** Do not share your private key with anyone!  It's the file that looks like `id_ed25519`.
```

File: docs/setup_git.md [hunk 3]
```diff
@@ -357,7 +360,7 @@ Branch 'main' set up to track remote branch 'main' from 'origin'.
 ```
 
 <div class="primer-spec-callout warning" markdown="1">
-**MacOS Pitfall:** If you are not prompted for your username and password, your credentials may be cached on your computer. [Clear your credentials from Keychain Access](https://docs.github.com/en/enterprise-server@3.4/get-started/getting-started-with-git/updating-credentials-from-the-macos-keychain#updating-your-credentials-via-keychain-access).
+**MacOS Pitfall:** If you are not prompted for your username and password when using the token, your credentials may be cached on your computer. [Clear your credentials from Keychain Access](https://docs.github.com/en/enterprise-server@3.4/get-started/getting-started-with-git/updating-credentials-from-the-macos-keychain#updating-your-credentials-via-keychain-access).
```
Comment (awdeorio) [inline docs/setup_git.md:363]:
```suggestion
**MacOS Pitfall:** If you are not prompted for your username and password when using a Personal Access Token, your credentials may be cached on your computer. [Clear your credentials from Keychain Access](https://docs.github.com/en/enterprise-server@3.4/get-started/getting-started-with-git/updating-credentials-from-the-macos-keychain#updating-your-credentials-via-keychain-access).
```

**Review by awdeorio — APPROVED** [review]
LGTM after a few suggestions.

**Comment by TheOmid** [comment]
Thanks for the suggestions, I've committed them into this PR

**Review by anishnya — APPROVED** [review]
LGTM

**Pull request merged**